android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildView.kt

```diff
@@ -73,159 +75,159 @@ fun HomeListChildView(
 	viewModel: HomeListChildViewModel,
 	uiState: HomeListUiState,
 ) {
-	if (uiState is HomeListUiState.Loading) {
-		Column(
-			modifier = modifier
-				.fillMaxSize(1.0f)
-				.imePadding(),
-			horizontalAlignment = Alignment.CenterHorizontally,
-			verticalArrangement = Arrangement.Center,
-		) {
-			CircularProgressIndicator()
-		}
-	} else if (uiState is HomeListUiState.Loaded) {
-		val filterType = remember { mutableStateOf(FilterType.LOCATION) }
-		val coroutineScope = rememberCoroutineScope()
-		val modelSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
-		var isBottomSheetOpen by remember {
-			mutableStateOf(false)
-		}
-		val listState: LazyListState = rememberLazyListState()
-		val lastItemIsShowing by remember {
-			derivedStateOf {
-				if (listState.layoutInfo.totalItemsCount == 0) {
-					false
-				} else {
-					listState.layoutInfo.visibleItemsInfo.lastOrNull()?.index == listState.layoutInfo.totalItemsCount - 1
+	Scaffold(
+		modifier = modifier.fillMaxSize(1.0f),
+		floatingActionButtonPosition = FabPosition.End,
+		floatingActionButton = {
+			FloatingActionButton(
+				modifier = Modifier.padding(
+					all = dimensionResource(id = R.dimen.zero),
+				),
+				onClick = {
+					viewModel.navHostController.navigate(NavigationDestination.CREATE_LISTING.fullNavPath)
+				},
+				shape = CircleShape,
+				containerColor = MaterialTheme.subletrPalette.subletrPink,
+				contentColor = MaterialTheme.subletrPalette.textOnSubletrPink,
+			) {
+				Text(
+					stringResource(id = R.string.plus_sign),
+					style = TextStyle(
+						fontSize = 24.sp
+					),
+					color = MaterialTheme.subletrPalette.textOnSubletrPink,
+				)
+			}
+		},
+		topBar = {
+			Box(modifier = Modifier)
+		},
+		bottomBar = {
+			Box(modifier = Modifier)
+		},
+	) { scaffoldPadding ->
+		if (uiState is HomeListUiState.Loading) {
+			Column(
+				modifier = Modifier
+					.fillMaxSize(1.0f)
+					.padding(paddingValues = scaffoldPadding)
+					.imePadding(),
+				horizontalAlignment = Alignment.CenterHorizontally,
+				verticalArrangement = Arrangement.Center,
+			) {
+				CircularProgressIndicator()
+			}
+		} else if (uiState is HomeListUiState.Loaded) {
+			val filterType = remember { mutableStateOf(FilterType.LOCATION) }
+			val coroutineScope = rememberCoroutineScope()
+			val modelSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+			var isBottomSheetOpen by remember {
+				mutableStateOf(false)
+			}
+			val listState: LazyListState = rememberLazyListState()
+			val lastItemIsShowing by remember {
+				derivedStateOf {
+					if (listState.layoutInfo.totalItemsCount == 0) {
+						false
+					} else {
+						listState.layoutInfo.visibleItemsInfo.lastOrNull()?.index == listState.layoutInfo.totalItemsCount - 1
+					}
```

Comment:
lastOrNull should give us null if there's nothing ... let's do a fancy kotlin oneliner :sunglasses:

Reply:
Huh? Not really sure how I'm supposed to make this into a fancy Kotlin oneliner?